lib/colorado_lottery.rb

```diff
@@ -0,0 +1,57 @@
+class ColoradoLottery
+  attr_reader :registered_contestants, :current_contestants, :winners
+
+  def initialize()
+    @registered_contestants = {}
+    @current_contestants = {}
+    @winners = []
+  end
+
+  def interested_and_18?(contestant, game)
+    if contestant.age >= 18
+      contestant_interest = false
+      if contestant.game_interests.each do |interest|
+        contestant_interest = true if interest == game.name
+      end
+      contestant_interest
+    end
+  else
+    false
+  end
+  end
+
+  def can_register?(contestant, game)
+    if interested_and_18?(contestant, game) == true && game.name == "Mega Millions" && contestant.age >= 18
+      true
+    elsif interested_and_18?(contestant, game) == true && game.name == "Cash 5" && contestant.state_of_residence == 'CO'
+      true
+    elsif interested_and_18?(contestant, game) == true && game.name == "Pick 4" && contestant.state_of_residence == 'CO'
+      true
+    else
+      false
+    end
```

Comment:
this doesn't quite work-- the statement is that the contestant either needs to live in state OR that the lottery is a national drawing. you are checking that the contestant lives in state, but never check the alternate condition for the lottery being a national drawing. 

this is also incredibly fragile, as you are manually checking each game at a time by looking at a list of hardcoded game names (which means if we ever added another game, this method would stop working). 

you could also use the method you wrote in your contestant class for `out_of_state` to avoid duplicating that logic here. 

```
def can_register?(contestant, game)
    return false unless interested_and_18?(contestant,game)

    !contestant.out_of_state? || game.national_drawing?
end 
```